javatests/arcs/core/data/expression/ExpressionTest.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data.expression
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.test.assertFailsWith
+
+/** Tests for [TypeVariable]. */
+@RunWith(JUnit4::class)
```

Comment:
I'd like to see test where the input and return of evalExpression are different.